src/node.cc

```diff
@@ -1510,15 +1511,16 @@ void AppendExceptionLine(Environment* env,
 
   Local<String> arrow_str = String::NewFromUtf8(env->isolate(), arrow);
 
-  if (!arrow_str.IsEmpty() && !err_obj.IsEmpty() && err_obj->IsNativeError()) {
+  if (!arrow_str.IsEmpty() && !err_obj.IsEmpty() &&
+      (!handlingFatalError || err_obj->IsNativeError())) {
```

Comment:
Would it work to drop the new `handling_fatal_error` part and do:

```
if (!arrow_str.IsEmpty() && !err_obj.IsEmpty()) {
  if (err_obj->IsNativeError()) {
    // err_obj->SetPrivate(...);
  }

  return;
}
```

That way the printing part would only be hit if allocation failed.